### PR TITLE
Activity result api update

### PR DESCRIPTION
**Pull request by ch8n**
## :scroll: Description
Migrated to new activity result API
https://developer.android.com/training/basics/intents/result

## :green_heart: How did you test it?
Manually Tested by exporting and Importing CSV files

### CODE REVIEW

**Comment by alisonthemonster** [comment]
Just gave this a test run on an emulator and ran into an issue :( 

- manually export CSV to downloads folder
- launch the import flow
- see your manually backed up entry as grey-ed out

I tried the same flow with develop and didn't run into this issue, here's a screenshot with the top two exports being from develop and the bottom one coming from this branch:
Screen Shot 2020-10-17 at 1 30 46 PM
Device is a pixel 2 on API 23.

I'm gonna look over the code once again to see if there's a MIME type missing or something like that!

File: app/src/main/java/journal/gratitude/com/gratitudejournal/ui/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -447,22 +402,38 @@ class SettingsFragment : PreferenceFragmentCompat(),
...
+        }
+
     /**
      * Opens the chooser to allow the user to select their CSV file.
      * */
     private fun selectCSVFile() {
         firebaseAnalytics.logEvent(LOOKED_FOR_DATA, null)
-
-        val intent = Intent(Intent.ACTION_GET_CONTENT)
-        intent.type = "text/csv|text/comma-separated-values|application/csv"
```
Comment (alisonthemonster) [inline app/src/main/java/journal/gratitude/com/gratitudejournal/ui/settings/SettingsFragment.kt:485]: I wonder if we need to keep this line?

File: app/src/main/java/journal/gratitude/com/gratitudejournal/ui/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -496,19 +467,43 @@ class SettingsFragment : PreferenceFragmentCompat(),
...
+
     /**
      * Opens the Storage Access Framework and lets the user select where they want to
      * export the CSV file.
      * */
     private fun createFileOnDevice() {
         val date = LocalDateTime.now().withNano(0).toString().replace(':', '-')
         val fileName = "PresentlyBackup$date.csv"
-        val intent = Intent(Intent.ACTION_CREATE_DOCUMENT).apply {
-            addCategory(Intent.CATEGORY_OPENABLE)
```
Comment (alisonthemonster) [inline app/src/main/java/journal/gratitude/com/gratitudejournal/ui/settings/SettingsFragment.kt:535]: I think we also need this one! Adding these in now and I'll post my results!

**Comment by alisonthemonster** [comment]
I believe this is how we can fix it!

```
class OpenCsvDocumentContract : ActivityResultContracts.OpenDocument() {
    override fun createIntent(context: Context, input: Array<out String>): Intent {
        val intent = super.createIntent(context, input)
        intent.type = "text/csv|text/comma-separated-values|application/csv"
        return intent
    }
}

class CreateCsvDocumentContract : ActivityResultContracts.CreateDocument() {
    override fun createIntent(context: Context, input: String): Intent {
        val intent = super.createIntent(context, input)
        return intent.apply {
            addCategory(Intent.CATEGORY_OPENABLE)
            type = "text/csv"
        }
    }
}
```
And then replace `OpenDocument` with `OpenCsvDocumentContract` and `CreateDocument` with `CreateCsvDocumentContract`!

**Comment by ch8n** [comment]
> I believe this is how we can fix it!
> 
> ```
> class OpenCsvDocumentContract : ActivityResultContracts.OpenDocument() {
>     override fun createIntent(context: Context, input: Array<out String>): Intent {
>         val intent = super.createIntent(context, input)
>         intent.type = "text/csv|text/comma-separated-values|application/csv"
>         return intent
>     }
> }
> 
> class CreateCsvDocumentContract : ActivityResultContracts.CreateDocument() {
>     override fun createIntent(context: Context, input: String): Intent {
>         val intent = super.createIntent(context, input)
>         return intent.apply {
>             addCategory(Intent.CATEGORY_OPENABLE)
>             type = "text/csv"
>         }
>     }
> }
> ```
> 
> And then replace `OpenDocument` with `OpenCsvDocumentContract` and `CreateDocument` with `CreateCsvDocumentContract`!

Oh yeah, I see also it's a cleaner approach thanks! I will test this and update PR

**Comment by alisonthemonster** [comment]
Sorry for the delay here! Ran into some rebase/merge issues with CI, but I'm thinking this will pass when it merges 🤞

**Pull request merged**